README.md

```diff
@@ -12,7 +12,8 @@ Variant's basic building blocks are:
 - `variant<Ts...>` - a type-safe representation for sum-types / discriminated unions
 - `recursive_wrapper<T>` - a helper type to represent recursive "tree-like" variants
 - `apply_visitor(visitor, myVariant)` - to invoke a custom visitor on the variant's underlying type
-- `get<T>()` - a function to directly unwrap a variant's underlying type
+- `bool is<T>()` - a function to test for a specific underlying type `T`
+- `T get<T>()` - a function to directly unwrap a variant's underlying type
```

Comment:
Should we document all of variant's api here? For example there is an unchecked get, a function valid, and others. Related to that: should we document that get may throw an exception on error?

(Btw not sure I'm the right person to review or make decisions here, all I did was implement the convenience match function and update the docs with some examples :P probably @artemp is good to tag for these kind of things)